src/util/lang_string.ts

```diff
@@ -46,16 +59,36 @@ const getSegmenter = () => {
  * @return {Array} array containing the graphemes
  */
 export const graphemeSplit = (textstring: string): string[] => {
-  segmenter || getSegmenter();
-  if (segmenter) {
-    const segments = segmenter.segment(textstring);
+  graphemeSegmenter || getGraphemeSegmenter();
+  if (graphemeSegmenter) {
+    const segments = graphemeSegmenter.segment(textstring);
     return Array.from(segments).map(({ segment }) => segment);
   }
 
   //Fallback
   return graphemeSplitImpl(textstring);
 };
 
+/**
+ * Divide a string into words
+ * @param {String} textstring String to split into words
+ * @param {RegExp} splitRegex Optional regex pattern for fallback splitting (default: /[ \t\r]/)
+ * @return {Array} array containing the words
+ */
+export const wordSplit = (
+  textstring: string,
+  splitRegex: RegExp
+): string[] => {
+  wordSegmenter || getWordSegmenter();
+  if (wordSegmenter) {
+    const segments = wordSegmenter.segment(textstring);
+    return Array.from(segments).map(({ segment }) => segment);
+  }
+
+  // Fallback to regex-based split
+  return textstring.split(splitRegex);
+};
```

Comment:
You may know, how this would work with languages that do not use spaces to divide words?
We have this naive approach in which we split by wordJoiners and then we assume a space was there.
the Intl.segmenter goes beyond that and knows how to split words that have no spaces, but then we are going to put a space back when we render text.

Did you encounter this issue?

Reply:
>  I do not think for the word splitter 'apple,banana' is 3 words

The Segmenter will also return the `isWordLike` value if granularity is `word` to be able to distinguish punctualisation from words if needed, an additional benefit of Segmenter